tests/Aspire.Hosting.Azure.Tests/AzureFunctionsTests.cs

```diff
@@ -312,21 +312,21 @@ param principalId string
               scope: funcstorage634f8
```

Comment:
Would setting `skipPreparer: false` in https://github.com/dotnet/aspire/blob/f5e596bb038ac396487da4350f78223ebdb9dd26/tests/Aspire.Hosting.Azure.Tests/AzureFunctionsTests.cs#L631 have helped catch this?

Reply:
It might have. But it wouldn't have been able to test what we needed it to, because these tests are targeting `WithAddAzureContainerAppsInfrastructure`, which needed to Execute all the startup hooks, not just the AzurePreparer.

Reply:
I've added a test that would have caught this bug.